packages/dd-trace/src/profiling/exporters/agent.js

```diff
@@ -1,11 +1,39 @@
 'use strict'
 
+const retry = require('retry')
+const { request } = require('http')
 const FormData = require('form-data')
 
+function sendRequest (options, body, callback) {
+  const req = request(options, res => {
+    if (res.statusCode >= 400) {
+      const error = new Error(`HTTP Error ${res.statusCode}`)
+      error.status = res.statusCode
+      callback(error)
+    } else {
+      callback(null, res)
+    }
+  })
+  req.on('error', callback)
+  if (body) req.write(body)
+  req.end()
+}
+
+function getBody (stream, callback) {
+  const chunks = []
+  stream.on('error', callback)
+  stream.on('data', chunk => chunks.push(chunk))
+  stream.on('end', () => {
+    callback(null, Buffer.concat(chunks))
+  })
+}
+
 class AgentExporter {
-  constructor ({ url, logger } = {}) {
+  constructor ({ url, logger, backoffTime, backoffTries } = {}) {
     this._url = url
     this._logger = logger
+    this._backoffTime = backoffTime || 1000
+    this._backoffTries = backoffTries || 6
```

Comment:
Should this be dynamic? For example, right now with a configured timeout of 1s, there would be a single retry, and for a configured timeout of 300s, there would be no more retries after the 6th one, thus needlessly waiting on an operation that will never succeed.

Reply:
Done. It's dynamic now. :)